### PR TITLE
Feature/collisions

### PR DESCRIPTION
**Pull request by carter-greis**

### CODE REVIEW

File: bin/main.cpp [hunk 1]
```diff
@@ -45,8 +45,8 @@ int main(int argc, char** argv)
       
     }
     if (focused) {
-        float deltaS = clock.restart().asSeconds();
-        deltaS = (deltaS < 1.f) ? 1.f : deltaS;
+        float deltaS = clock.restart().asMilliseconds();
+        deltaS = (deltaS > 1.f) ? 1.f : deltaS;
```
Comment (jkula25) [inline bin/main.cpp]: I thought we agreed we were gonna leave this as seconds, because milliseconds screws everything up

File: include/AIView.h [hunk 2]
```diff
@@ -7,9 +7,9 @@
 
 class AIView: public View {
 	private:        
-		
+		int gameState;
```
Comment (jkula25) [inline include/AIView.h]: Why is gameState a member variable if it is going to be dynamic throughout the game?

File: include/GameLogic.h [hunk 3]
```diff
@@ -2,8 +2,13 @@
 #define GAMELOGIC_H
 #include <SFML/Graphics.hpp>
 #include <vector>
+#include <functional>
 #include "player.h"
 #include "platform.h"
+#include "spike.h"
+#include "projectile.h"
+#include "enemy.h"
+#include "spike.h"
```
Comment (jkula25) [inline include/GameLogic.h]: You included spike.h twice, delete one

File: src/AIView.cpp [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
-#include "GameLogic.h"
 #include "AIView.h"
 #include <iostream>
 
-AIView::AIView(GameLogic& logic): View(logic) {}
+//AIView::AIView(GameLogic& logic): View(logic) {}
 
-void AIView::init() {}
+void AIView::init() {
+    gameState = logic.getGameState();
```
Comment (jkula25) [inline src/AIView.cpp]: We should not set gameState when initializing the AIView because the game state is dynamic

File: src/AIView.cpp [hunk 5]
```diff
@@ -1,17 +1,23 @@
 #include <SFML/Graphics.hpp>
 #include <SFML/Window.hpp>
-#include "GameLogic.h"
 #include "AIView.h"
 #include <iostream>
 
-AIView::AIView(GameLogic& logic): View(logic) {}
+//AIView::AIView(GameLogic& logic): View(logic) {}
```
Comment (jkula25) [inline src/AIView.cpp]: If this is not needed, delete it

File: src/AIView.cpp [hunk 6]
```diff
@@ -1,17 +1,23 @@
...
 
-void AIView::init() {}
+void AIView::init() {
+    gameState = logic.getGameState();
+}
 
 void AIView::update(float time) {
     //only active if game is running
-    if ( logic.getGameState() == 1 ) {
+    switch ( gameState ) {
```
Comment (jkula25) [inline src/AIView.cpp]: Change gameState to logic.getGameState() since the game state is dynamic

File: src/GameLogic.cpp [hunk 7]
```diff
@@ -60,23 +87,23 @@ Player& GameLogic::getPlayer() {
     return player;
 }
 
-std::vector<Actor>& GameLogic::getActors() {
+std::vector<std::reference_wrapper<Actor>> GameLogic::getActors() {
```
Comment (jkula25) [inline src/GameLogic.cpp]: Do we still need to return the actual reference to the actors vector, or does just a copy suffice?

File: src/platform.cpp [hunk 8]
```diff
@@ -16,7 +16,9 @@ void Platform::init(const float w, const float h, const float x, const float y)
     {
         // error...
     }
+    texture.setRepeated(true);
     setSprite(sf::Sprite(texture));
-    getSprite().setScale(sf::Vector2f(scaleWidth, scaleHeight));
+    getSprite().setTextureRect({ 0, 0, (texture.getSize().x) * scaleWidth, (texture.getSize().y) * scaleHeight });
+    //getSprite().setScale(sf::Vector2f(scaleWidth, scaleHeight));
```
Comment (jkula25) [inline src/platform.cpp:22]: If this is not needed, delete this comment

File: bin/main.cpp (hunk 1 above)
Comment (carter-greis) [reply bin/main.cpp]: I can't remember, this was just what was working last when I committed it.

Comment (carter-greis) [reply bin/main.cpp]: I changed it back to seconds

File: src/GameLogic.cpp (hunk 7 above)
Comment (carter-greis) [reply src/GameLogic.cpp]: I think a reference to it will use less memory overhead, I am not sure though.

File: src/platform.cpp (hunk 8 above)
Comment (carter-greis) [reply src/platform.cpp:22]: I think we should leave it just in case we need to change it back later for some reason

**Pull request merged**